pkgdb/include/flox/lock-flake-installable.hh

```diff
@@ -57,16 +57,19 @@ public:
 
 struct LockedInstallable
 {
-  std::string                             lockedUrl;
-  std::optional<std::string>              flakeDescription;
-  std::string                             lockedFlakeAttrPath;
-  std::string                             derivation;
-  std::map<std::string, std::string>      outputs;
-  std::vector<std::string>                outputNames;
-  std::optional<std::set<std::string>>    outputsToInstall;
-  std::optional<std::set<std::string>>    requestedOutputsToInstall;
-  std::string                             packageSystem;
-  std::string                             lockedSystem;
+  std::string                          lockedUrl;
+  std::optional<std::string>           flakeDescription;
+  std::string                          lockedFlakeAttrPath;
+  std::string                          derivation;
+  std::map<std::string, std::string>   outputs;
+  std::vector<std::string>             outputNames;
+  std::optional<std::set<std::string>> outputsToInstall;
+  std::optional<std::set<std::string>> requestedOutputsToInstall;
+  /** The system the package reports in <drv>.system */
+  std::string packageSystem;
```

Comment:
Question: in what scenario are `packageSystem` and `system` different? I'm interpreting that to mean that `pkgdb` has requested to lock for one system and the package declares itself as a different system, in which case I don't see how that would be a successful lock operation.

Reply:
Yeah that's right. We could start erroring, see discussion in https://github.com/flox/flox/pull/1752#discussion_r1671373984